modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_8_0/UpgradeDDMStructure.java

```diff
@@ -0,0 +1,399 @@
+						).build());
+
+		DDMFormLayout ddmFormLayout =
+			ddmFormLayoutDeserializerDeserializeResponse.getDDMFormLayout();
+
+		for (DDMFormLayoutPage ddmFormLayoutPage :
+				ddmFormLayout.getDDMFormLayoutPages()) {
+
+			List<DDMFormLayoutRow> ddmFormLayoutRows =
+				ddmFormLayoutPage.getDDMFormLayoutRows();
+
+			for (DDMFormLayoutRow ddmFormLayoutRow : ddmFormLayoutRows) {
+				List<DDMFormLayoutColumn> ddmFormLayoutColumns =
+					ddmFormLayoutRow.getDDMFormLayoutColumns();
+
+				for (DDMFormLayoutColumn ddmFormLayoutColumn :
+						ddmFormLayoutColumns) {
+
+					List<String> ddmFormFieldNames =
+						ddmFormLayoutColumn.getDDMFormFieldNames();
+
+					if (ddmFormFieldNames.size() > 1) {
+						String ddmFormFieldName = _generateFieldSetName(
+							ddmForm);
+
+						tuples.add(
+							new Tuple(
+								ddmFormFieldName, "Fields Group",
+								ddmFormLayoutColumn.getSize(),
+								ddmFormFieldNames));
+
+						ddmFormLayoutColumn.setDDMFormFieldNames(
+							Arrays.asList(ddmFormFieldName));
+					}
+				}
+			}
+		}
+
+		if (!tuples.isEmpty()) {
+			_nestedFieldsMap.put(structureVersionId, tuples);
+		}
+
+		DDMFormLayoutSerializerSerializeResponse
+			ddmFormLayoutSerializerSerializeResponse =
+				_ddmFormLayoutSerializer.serialize(
+					DDMFormLayoutSerializerSerializeRequest.Builder.newBuilder(
+						ddmFormLayout
+					).build());
+
+		return ddmFormLayoutSerializerSerializeResponse.getContent();
+	}
+
+	private void _upgradeDDMStructureVersion() throws Exception {
+		StringBundler sb = new StringBundler(5);
+
+		sb.append("select DDMStructureVersion.structureVersionId, ");
+		sb.append("DDMStructureVersion.definition from DDMStructure inner ");
+		sb.append("join DDMStructureVersion on DDMStructure.structureId = ");
+		sb.append("DDMStructureVersion.structureId where ");
+		sb.append("DDMStructure.classnameId = ?");
+
+		try (PreparedStatement ps1 = connection.prepareStatement(sb.toString());
+			PreparedStatement ps2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructureVersion set definition = ? where " +
+						"structureVersionId = ?")) {
+
+			ps1.setLong(
+				1, PortalUtil.getClassNameId(DDMFormInstance.class.getName()));
+
+			try (ResultSet rs = ps1.executeQuery()) {
+				while (rs.next()) {
+					String definition = rs.getString("definition");
+
+					long structureVersionId = rs.getLong("structureVersionId");
+
+					ps2.setString(
+						1,
+						_upgradeDDMStructureVersionDefinition(
+							definition, structureVersionId));
+
+					ps2.setLong(2, structureVersionId);
+
+					ps2.addBatch();
+				}
+
+				ps2.executeBatch();
+			}
+		}
+	}
+
+	private String _upgradeDDMStructureVersionDefinition(
+		String definition, long structureVersionId) {
+
+		DDMForm ddmForm = _getDDMForm(definition);
+
+		Map<String, DDMFormField> ddmFormFieldsMap =
+			ddmForm.getDDMFormFieldsMap(false);
+
+		List<Tuple> tuples = _nestedFieldsMap.get(structureVersionId);
```

Comment:
@marcosapmf, variables of type `Tuple` aren't named only as `tuples` in Liferay code, usually, they're named as `entityTuples`, like `_serviceReferenceServiceTuples`. Can you please change it?

cc @carolmariaabb